docs/content/preview/architecture/transactions/transaction-priorities.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Transaction priorities
+headerTitle: Transaction priorities
+linkTitle: Transaction priorities
+description: Details about Transaction priorities in YSQL
+menu:
+  preview:
+    identifier: architecture-transaction-priorities
+    parent: architecture-acid-transactions
+    weight: 1153
+type: docs
+---
+
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+
+  <li >
+    <a href="../transaction-priorities/" class="nav-link active">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL
+    </a>
+  </li>
+
+</ul>
+
+When using the [Fail-on-Conflict](../concurrency-control/#fail-on-conflict) concurrency control policy, transactions are assigned priorities that help decide which transactions should be aborted in case of conflict.
+
+There are two priority buckets, each having a priority range of [reals](https://www.postgresql.org/docs/current/datatype.html) in [0, 1] as follows:
+
+1. `High-priority` bucket: if the first statement in a transaction takes a `FOR UPDATE/ FOR SHARE/ FOR NO KEY UPDATE` explicit row lock using SELECT, it will be assigned a priority from this bucket.
+
+2. `Normal-priority` bucket: all other transactions are assigned a priority from this bucket.
+
+Note that a transaction with any priority P1 from the high-priority bucket can abort a transaction with any priority P2 from the normal-priority bucket. For example, a transaction with priority 0.1 from the high-priority bucket can abort a transaction with priority 0.9 from the normal-priority bucket.
+
+Priorities are randomly chosen from the applicable bucket. However, there are two user configurable session variables that can help control the priority assigned to transaction is a specific session: `yb_transaction_priority_lower_bound` and `yb_transaction_priority_upper_bound`. These variables help set lower and upper bounds on the randomly assigned priority that a transaction should receive from the applicable bucket. These variables accept a value of `real` datatype in the range [0, 1]. Also note that the same bounds apply to both buckets.
+
+{{< note title="All single shard transactions have a priority of 1 in the normal-priority bucket." >}}
+{{</note >}}
+
+The `yb_get_current_transaction_priority` function can be used to fetch the transaction priority of the current active transaction. It outputs a pair `<priority> (bucket)`, where `<priority>` is of a real datatype between [0, 1] with 9 decimal units of precision, and `<bucket>` is either `Normal` or `High`.
+
+{{< note title="Note">}}
+As an exception, if a transaction is assigned the highest priority possible, that is, a priority of 1 in the high-priority bucket, then a single value _highest priority transaction_ is returned without any real value.
+{{</note >}}
+
+A transaction's priority is `0.000000000 (normal-priority transaction)` until a transaction is really started.
```

Comment:
what does "really started" mean?

Reply:
a distributed transaction

Reply:
we don't start a transaction until a write is performed